crates/ra_assists/src/handlers/change_return_type_to_result.rs

```diff
@@ -0,0 +1,946 @@
+use ra_syntax::{
+    ast, AstNode,
+    SyntaxKind::{COMMENT, WHITESPACE},
+    SyntaxNode, TextSize,
+};
+
+use crate::{Assist, AssistCtx, AssistId};
+use ast::{BlockExpr, Expr, LoopBodyOwner};
+
+// Assist: change_return_type_to_result
+//
+// Change the function's return type to Result.
+//
+// ```
+// <|>fn foo() -> i32 { 42i32 }
+// ```
+// ->
+// ```
+// fn foo() -> Result<i32, > { Ok(42i32) }
+// ```
+pub(crate) fn change_return_type_to_result(ctx: AssistCtx) -> Option<Assist> {
+    let fn_def = ctx.find_node_at_offset::<ast::FnDef>();
+    let fn_def = &mut fn_def?;
+    if fn_def.ret_type().is_none() {
+        return None;
+    }
+    let ret_type = &fn_def.ret_type().unwrap().type_ref()?;
```

Comment:
`is_none` / `is_some` check followed buy unwrap is always worse than match. Such checks should always be refactored to use `match` / `if let`.